core/drivers/stm32_iwdg.c

```diff
@@ -355,50 +335,30 @@ static TEE_Result stm32_iwdg_setup(struct stm32_iwdg_device *iwdg,
 	return TEE_SUCCESS;
 }
 
-static TEE_Result stm32_iwdg_register(struct stm32_iwdg_device *iwdg)
-{
-	TEE_Result res = TEE_ERROR_GENERIC;
-
-	if (is_assigned_to_nsec(iwdg)) {
-		stm32mp_register_non_secure_periph_iomem(iwdg->base.pa);
-	} else {
-		stm32mp_register_secure_periph_iomem(iwdg->base.pa);
-
-		/* Expose watchdog runtime service only to secure IWDG */
-		iwdg->wdt_chip.ops = &stm32_iwdg_ops;
-
-		res = watchdog_register(&iwdg->wdt_chip);
-		if (res)
-			return res;
-	}
-
-	SLIST_INSERT_HEAD(&iwdg_dev_list, iwdg, link);
-
-	return TEE_SUCCESS;
-}
-
 static TEE_Result stm32_iwdg_probe(const void *fdt, int node,
 				   const void *compat_data __unused)
 {
 	struct stm32_iwdg_device *iwdg = NULL;
 	TEE_Result res = TEE_SUCCESS;
 
+	static_assert(IS_ENABLED(CFG_WDT) && IS_ENABLED(CFG_WDT_SM_HANDLER));
```

Comment:
This is not useful as CFG_STM32_IWDG is disabled when either CFG_WDT or CFG_WDT_SM_HANDLER is disabled?

Reply:
It is checked in platform config but i think it's better to assert it locally to make it clear at driver level.
Overkilling?

Reply:
I think it's a bit too much :wink:

Reply:
ok